### PR TITLE
BAVL-342 changes to handle unknown courts and probation teams.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Adds support to the migration process to deal with uknown courts and/or probation teams.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/migration/MigrateVideoBookingService.kt [hunk 1]
```diff
@@ -58,9 +60,13 @@ class MigrateVideoBookingService(
...
     }
 
-    // TODO still need to migrate if court not found
-    val court = mappingService.mapCourtCodeToCourt(bookingToMigrate.courtCode!!)
-      ?: throw NullPointerException("Court not found for code ${bookingToMigrate.courtCode} for court booking ${bookingToMigrate.videoBookingId}")
+    val court = if (bookingToMigrate.courtCode != null) {
+      mappingService.mapCourtCodeToCourt(bookingToMigrate.courtCode)
+        ?: throw NullPointerException("Court not found for code ${bookingToMigrate.courtCode} for court booking ${bookingToMigrate.videoBookingId}")
+    } else {
+      // TODO this assumes we have a court name.  Where does this go/get recorded, could potentially include/add to the comments?
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/migration/MigrateVideoBookingService.kt:67]: Thoughts?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/migration/MigrateVideoBookingService.kt [hunk 2]
```diff
@@ -121,9 +127,13 @@ class MigrateVideoBookingService(
...
       )
 
-    // TODO still need to migrate if probation not found
-    val probationTeam = mappingService.mapProbationTeamCodeToProbationTeam(bookingToMigrate.courtCode!!)
-      ?: throw NullPointerException("Probation team not found for code ${bookingToMigrate.courtCode} for probation booking ${bookingToMigrate.videoBookingId}")
+    val probationTeam = if (bookingToMigrate.courtCode != null) {
+      mappingService.mapProbationTeamCodeToProbationTeam(bookingToMigrate.courtCode)
+        ?: throw NullPointerException("Probation team not found for code ${bookingToMigrate.courtCode} for probation booking ${bookingToMigrate.videoBookingId}")
+    } else {
+      // TODO this assumes we have a probation team name.  Where does this go/get recorded, could potentially include/add to the comments?
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/migration/MigrateVideoBookingService.kt:134]: Thoughts?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/migration/MigrateVideoBookingService.kt (hunk 1 above)
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/migration/MigrateVideoBookingService.kt:67]: Yes -  looks ok. Provided we cater for an UNKNOWN court in the data.

File: src/main/resources/migrations/common/V2024.04.23__baseline_data.sql [hunk 3]
```diff
@@ -384,7 +384,8 @@ values (1, 'DRBYMC', 'Derby Justice Centre', true, null, 'TIM', current_timestam
 (323, 'PORTFC', 'Port Talbot Family', true, null, 'TIM', current_timestamp),
 (324, 'SWANCV', 'Swansea Civil', true, null, 'TIM', current_timestamp),
 (325, 'SWANFC', 'Swansea Family', true, null, 'TIM', current_timestamp),
-(326, 'ABERMC', 'Aberystwyth Magistrates', true, null, 'TIM', current_timestamp);
+(326, 'ABERMC', 'Aberystwyth Magistrates', true, null, 'TIM', current_timestamp),
+(327, 'UNKNOWN', 'Unknown Court', false, 'When the court is not known.', 'MATT', current_timestamp);
```
Comment (timharrison-moj) [inline src/main/resources/migrations/common/V2024.04.23__baseline_data.sql:388]: And here it is.

**Pull request merged**